src/App.tsx

```diff
@@ -15,47 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SortFieldName = 'Name',
+  SortFieldLength = 'Length',
+}
+
+function getPrepearedGoods(goods: string[],
+  sortType: string, isReversed: boolean) {
+  const prepearedGoods = [...goods];
+
+  if (sortType !== '') {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.SortFieldName: {
+          return good1.localeCompare(good2);
+        }
+
+        case SortType.SortFieldLength: {
+          return good1.length - good2.length;
+        }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? prepearedGoods.reverse() : prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const vissibleGoods = getPrepearedGoods(goodsFromServer,
+    sortType, isReversed);
+
+  const reverse = () => {
+    setIsReversed(prev => !prev);
+  };
+
+  const isStateNotReversedOrSorted
+    = isReversed || sortType;
```

Comment:
The variable 'isStateNotReversedOrSorted' is misleading, since it's true when the state IS reversed or sorted. Consider renaming it to 'isStateReversedOrSorted'.